pkg/utils/netlink.go

```diff
@@ -108,3 +119,40 @@ func bridgeTrunkVlanAdd(nlink Netlink, link netlink.Link, vlans []int) error {
 	}
 	return nil
 }
+
+// GetBridgeByLink returns linux bridge link if provided link belongs to any
+// if provided link is a part of a bond will return a bridge to which this bond belongs to
+func GetBridgeByLink(nLink Netlink, link netlink.Link) (netlink.Link, error) {
+	master, err := getMasterInterface(nLink, link)
+	if err != nil {
+		return nil, err
+	}
+	if master == nil {
+		return nil, nil
+	}
+	switch master.Type() {
+	case linkTypeBridge:
+		return master, nil
+	case linkTypeBond:
+		bondMaster, err := getMasterInterface(nLink, master)
+		if err != nil {
+			return nil, err
+		}
+		if bondMaster != nil && bondMaster.Type() == linkTypeBridge {
+			return bondMaster, nil
+		}
+		// master for bond interface not found, or master is not a bridge
+		return nil, nil
+	default:
+		// unknown master type
+		return nil, nil
+	}
+}
+
+// getMasterInterface returns a master interface for the link if it exists
+func getMasterInterface(nLink Netlink, link netlink.Link) (netlink.Link, error) {
+	if link.Attrs().MasterIndex == 0 {
+		return nil, nil
```

Comment:
you can return that master index if not found error. (related to comments above)

Reply:
i changed the function to return errors